feat: option to configure partition with env var

mbergkvist opened the pull request:
Adds support for deployment to aws-cn partition by checking an optional environemt variable, `CDK_AWS_PARTITION`. The partition is currently not available to the pipeline in the `jobForDeploy`, adding the environment variable is a non-breaking change to get the pipeline to deploy to aws-cn.

This has been verified with stacks and nested stacks in multiple stages in aws-cn.

Fixes #820

mbergkvist commented:
#868

kaizencc reviewed (changes requested):
looks good, can we unit test this just for posterity?

mbergkvist commented:
> looks good, can we unit test this just for posterity?

Absolutely. I'll see what I can do.

> also, maybe a reference in the README would be helpful. doesn't have to be front and center, i don't imagine that many people in aws-cn are using this :)

Of course. I don't know why I didn't add it in the first place 🤦 
Any suggestion under what section this would belong?

mbergkvist commented:
@kaizencc I've updated the documentation, I'm not a technical writer though so feel free to update it.

I've also added test suites, I figured it would get good coverage by duplicating the current test suites that are relevant for the changes here, and tweaking them for the aws-cn case.

mbergkvist commented:
@kaizencc Is this ok to be merged?

kaizencc commented on README.md:
```diff
@@ -644,6 +645,19 @@ const pipeline = new GitHubWorkflow(app, 'Pipeline', {
+
+The `CDK_AWS_PARTITION` environment variable can be used to specify the AWS partition for the pipeline.
+If it's specified to `aws-cn`, the assets generated by pipeline will reference the resources in
+`.amazonaws.com.cn` instead of `.amazonaws.com.cn`.
```
```suggestion
`.amazonaws.com.cn` instead of `.amazonaws.com`.
```

mbergkvist replied:
🤦‍♂️
